### ISSUE TITLE
Name?

### ISSUE THREAD

**Issue by eric-m** [Apostolique/Agar.io-bot#367]
Is there a way to retrieve enemy names?

I'm playing around on my own fork, and I'd like to create a larger danger area when two like-named blobs are less than size1 + size2 apart so the bot knows to high tail it away.  :)

Thanks for creating this.  I've been wanting to get better at JavaScript.  My daughter thinks this is great, so it's a great way to teach her programming and figure out JavaScript as I go.  I'm an old (emphasis on old :) C++/Java programmer.

**Comment by Apostolique** [comment]
Hey! Nice to see you like the project. JavaScript isn't that hard to learn, but if you come from the C++ / Java programming world (I also started with those languages.) you're gonna hate the debugging tools and the way programs can crash. /rant

Every cell object has a .name property:
https://github.com/Apostolique/Agar.io-bot/blob/b6bd1d2c6c7b48d734e181e67c31606d4ed230c7/launcher.user.js#L1939

**Comment by eric-m** [comment]
Thanks.  Yeah, I'm not a fan of runtime errors and debugging.  On the positive side, it meant I could return an additional parameter with ease.  :)

Is there a color attribute obfuscated in there somewhere?  I've managed to identify duplicate player names (other than the unnamed cells), and then sum of the size of the split cells, add that to the normal and split danger sizes.  It now draws slightly bigger circles around split cells.  It doesn't yet differentiate between just split and about to rejoin.  The problem I have now is the two Doge's on the same screen appear as split cells to me.  :)  Figure if I can add a color element, I'll reduce errors.

**Comment by Apostolique** [comment]
If you're talking about the color a cell can have, every cell has a color property. cell.color

**Comment by eric-m** [comment]
Thanks.  That seemed like a "duh" question.  :)

I think I have my split cell recognition working now.  If you're interested, I could try to push it back into your code base. 

It filters out food and viruses, and then uses a combination of name & color to determine split cells.

It isn't optimized yet.  A cluster of split cells is seen as dangerous if the aggregate size would be a threat.  Next up is tweaking it to increase the danger size once it looks like the split cells are indeed merging back, and if there is some separation between them, treat them like food if they are small enough individually.

**Comment by Apostolique** [comment]
If you have the code in your repo, I would gladly test it.

**Comment by eric-m** [comment]
https://github.com/eric-m/Agar.io-bot

**Comment by eric-m** [comment]
Actually...I broke it somehow.  Don't bother looking at it quite yet.  :)

**Comment by Apostolique** [comment]
Okay :)

**Comment by eric-m** [comment]
It's back to working.  I think it broke when I started return the array from a function instead of passing it as an argument.  I had valid values in it.  I returned it.  I no longer had valid values.  I changed it so I created an empty array, passed it into a function as an argument, and then I had valid values in both places.  

My version seems to grow slightly larger and last for slightly longer times.  I added an A/B toggle so I can run both versions in different windows.  Next step is I want to figure out how to start logging massive amounts of data efficiently so can better understand what tweaks work better.  :)

If I haven't worn out my welcome of asking trivial questions...  How do I get a linefeed in text returned from displayText?

https://github.com/eric-m/Agar.io-bot/blob/master/bot.user.js#L139
